pkg/cmd/add_workspace.go

```
@@ -0,0 +1,97 @@
+package cmd
+
+import (
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/ZupIT/ritchie-cli/pkg/formula"
+	"github.com/ZupIT/ritchie-cli/pkg/prompt"
+)
+
+const (
+	workspaceNameFlag            = "name"
+	workspaceNameFlagDescription = "Workspace name"
+	workspacePathFlag            = "path"
+	workspacePathFlagDescription = "Workspace path"
+)
+
+type addWorkspaceCmd struct {
+	workspace formula.WorkspaceAddLister
+	inText    prompt.InputText
+}
+
+func NewAddWorkspaceCmd(
+	workspace formula.WorkspaceAddLister,
+	inText prompt.InputText,
+) *cobra.Command {
+	a := addWorkspaceCmd{
+		workspace: workspace,
+		inText:    inText,
+	}
+
+	cmd := &cobra.Command{
+		Use:       "workspace",
+		Short:     "Add new workspace",
+		Example:   "rit add workspace",
+		RunE:      a.runPrompt(),
+		ValidArgs: []string{""},
+		Args:      cobra.OnlyValidArgs,
+	}
+
+	flags := cmd.Flags()
+	flags.String(workspaceNameFlag, "", workspaceNameFlagDescription)
+	flags.String(workspacePathFlag, "", workspacePathFlagDescription)
+
+	return cmd
+}
+
+func (a addWorkspaceCmd) runPrompt() CommandRunnerFunc {
+	return func(cmd *cobra.Command, args []string) error {
+		if hasFlags, workspaceByFlags := hasFlags(cmd); hasFlags {
+			return a.workspace.Add(workspaceByFlags)
+		}
+
+		workspaceByPrompt, err := addWorkspaceFromPrompt(a)
+		if err != nil {
+			return err
+		}
+
+		return a.workspace.Add(workspaceByPrompt)
+	}
+}
+
+func hasFlags(cmd *cobra.Command) (bool, formula.Workspace) {
+	workspaceName, _ := cmd.Flags().GetString(workspaceNameFlag)
+	workspacePath, _ := cmd.Flags().GetString(workspacePathFlag)
+
+	if len(workspaceName) == 0 || len(workspacePath) == 0 {
+		return false, formula.Workspace{}
+	}
```

I think we should return an error if the user has any flags but forgets to fill them instead of falling back to prompt